src/gap_to_julia.jl

```diff
@@ -78,164 +94,153 @@ function gap_to_julia(::Type{Rational{T}}, x::Int64) where T <: Integer
-    if Globals.IsStringRep( obj )
-        array = UNSAFE_CSTR_STRING( obj )
-        return deepcopy(array)
-    elseif Globals.IsList( obj )
-        return Array{UInt8,1}(map(i->gap_to_julia(UInt8,obj[i]),1:length(obj)))
-    else
-        throw(ArgumentError("<obj> is not a list"))
-    end
+## Convert GAP string to Vector{UInt8} (==Array{UInt8,1})
+function gap_to_julia(::Type{Vector{UInt8}}, obj::GapObj)
+    Globals.IsStringRep(obj) && return deepcopy(UNSAFE_CSTR_STRING(obj))
+    Globals.IsList(obj) && return [gap_to_julia(UInt8,obj[i]) for i in 1:length(obj)]
+    throw(ConversionError(obj, Vector{UInt8}))
 end
 
 ## BitArrays
-function gap_to_julia( ::Type{BitArray{1}}, obj :: GapObj )
-    if ! Globals.IsBlist( obj )
-        throw(ArgumentError("<obj> is not a boolean list"))
-    end
+function gap_to_julia(::Type{BitArray{1}}, obj::GapObj)
+    ! Globals.IsBlist(obj) && throw(ConversionError(obj, BitArray{1}))
     len = Globals.Length( obj )
-    result = BitArray( undef, len )
+    result = BitArray(undef, len)
     for i in 1:len
       result[i] = obj[i]
     end
     return result
 end
 
-## Arrays
-function gap_to_julia( ::Type{Array{Obj,1}}, obj :: GapObj, recursion_dict = IdDict() )
-    if ! Globals.IsList( obj )
-        throw(ArgumentError("<obj> is not a list"))
-    end
-    if haskey(recursion_dict,obj)
-        return recursion_dict[obj]
-    end
-    len_list = length(obj)
-    new_array = Array{Any,1}( undef, len_list)
-    recursion_dict[obj] = new_array
-    for i in 1:len_list
-        new_array[i] = ElmList(obj, i)  # returns 'nothing' for holes in the list
-    end
-    return new_array
-end
-
-function gap_to_julia( ::Type{Array{T,1}}, obj :: GapObj, recursion_dict = IdDict() ) where T
-    if ! Globals.IsList( obj )
-        throw(ArgumentError("<obj> is not a list"))
-    end
-    if haskey(recursion_dict,obj)
-        return recursion_dict[obj]
-    end
-    len_list = length(obj)
-    new_array = Array{T,1}( undef, len_list)
-    recursion_dict[obj] = new_array
-    for i in 1:len_list
-        current_obj = ElmList(obj, i)
-        new_array[i] = get!(recursion_dict, current_obj) do
-            gap_to_julia(T, current_obj, recursion_dict)
+## Vectors
+function gap_to_julia(::Type{Vector{T}}, obj::GapObj, recursion_dict=IdDict(); recursive=true) where T
+    ! Globals.IsList(obj) && throw(ConversionError(obj, Vector{T}))
+    if ! haskey(recursion_dict, obj)
+        len_list = length(obj)
+        new_array = Vector{T}(undef, len_list)
+        recursion_dict[obj] = new_array
+        for i in 1:len_list
+            current_obj = ElmList(obj, i)  # returns 'nothing' for holes in the list
+            if recursive
+                new_array[i] = get!(recursion_dict, current_obj) do
+                    gap_to_julia(T, current_obj, recursion_dict;
+                                 recursive=true)
+                end
+            else
+                new_array[i] = current_obj
```

Comment:
does this even work with `current_obj = nothing`?
```julia
julia> x = Vector{Int}(undef, 3)
3-element Array{Int64,1}:
 139983047916960
 139983047177608
 139983047843440

julia> x[1] = nothing
ERROR: MethodError: Cannot `convert` an object of type Nothing to an object of type Int64
Closest candidates are:
  convert(::Type{Int64}, ::Polymake.Integer) at /home/kalmar/.julia/dev/Polymake/src/convert.jl:55
  convert(::Type{Int64}, ::Polymake.Rational) at /home/kalmar/.julia/dev/Polymake/src/convert.jl:58
  convert(::Type{JT}, ::CT) where {JT<:Number, CT<:Union{CxxWrap.CxxWrapCore.CxxSigned, CxxWrap.CxxWrapCore.CxxUnsigned}} at /home/kalmar/.julia/packages/CxxWrap/LfTHV/src/CxxWrap.jl:114
  ...
Stacktrace:
 [1] setindex!(::Array{Int64,1}, ::Nothing, ::Int64) at ./array.jl:826
 [2] top-level scope at REPL[14]:1
```

Reply:
A list with holes cannot be converted into an Array{Int64,1} unambiguously, so triggering an error is IMHO the only correct action. Of course we  still might want to test foe it explicitly and throw a dedicated error, but that's a minor detail in my eyes.

Of course one can define conversions for this case, eg "skip all holes" (can be achieved by calling GAP's `Compacted` function before the conversion), but this are not "natural" (e.g. array indices change), so I don't think it's a good idea to use them automatically